cranelift/fuzzgen/src/function_generator.rs

```diff
@@ -229,6 +237,16 @@ where
         Ok((block, args))
     }
 
+    /// Valid blocks for jump tables have to have no parameters in the signature, and must also
+    /// not be the first block.
+    fn generate_valid_jumptable_target_blocks(&mut self) -> Vec<Block> {
+        self.blocks[1..]
+            .iter()
+            .filter(|(_, sig)| sig.len() == 0)
+            .map(|(b, _)| *b)
+            .collect()
```

Comment:
The "empty blockparams" condition is of course necessary, but how do we guard against cases where no blocks fit this requirement, and we end up doing a `choose()` on an empty list (which will panic)? Possibly I'm just missing this invariant elsewhere -- if so, could you add a comment describing why it'll always succeed, and maybe an `assert!(target_blocks.len() > 0)` or somesuch?

Reply:
This already happens all the time. i.e. we are never guaranteed to generate variables of any type, so any instruction that requests a value of a type that doesn't exist ends up doing a choose on an empty block.

https://github.com/bytecodealliance/wasmtime/blob/3db8043a0284dd40e2859f8df3e05e730a706624/cranelift/fuzzgen/src/function_generator.rs#L194-L198

What ends up happening here, is that choose returns an error, we bubble the error up and into cargo-fuzz, which marks this input as invalid and starts another round of fuzzing.

Reply:
Ah, great, the key bit is that the failure bubbles up; I couldn't recall if `choose()` just panics in this case instead. All good then -- I'll go ahead and merge, thanks!